### ISSUE TITLE
A property that has anyOf key is ignored

### ISSUE THREAD

**Issue by SokichiFujita** [apple/swift-openapi-generator#203]
I generated the following OpenAPI specification that has the `model` property in request object.

https://github.com/openai/openai-openapi/blob/06d545011a2fbe4533398cc2f13b33f6beba5f09/openapi.yaml#L2117

```yaml
    CreateChatCompletionRequest:
      type: object
      properties:
        model:
          description: ID of the model to use. See the [model endpoint compatibility](/docs/models/model-endpoint-compatibility) table for details on which models work with the Chat API.
          example: "gpt-3.5-turbo"
          anyOf:
            - type: string
            - type: string
              enum:
                [
                  "gpt-4",
                  "gpt-4-0314",
                  "gpt-4-0613",
                  "gpt-4-32k",
                  "gpt-4-32k-0314",
                  "gpt-4-32k-0613",
                  "gpt-3.5-turbo",
                  "gpt-3.5-turbo-16k",
                  "gpt-3.5-turbo-0301",
                  "gpt-3.5-turbo-0613",
                  "gpt-3.5-turbo-16k-0613",
                ]
          x-oaiTypeLabel: string
        messages:

```

However after building, `Types.swift` does not have the `model` key.

```swift
            public init(
                messages: [Components.Schemas.ChatCompletionRequestMessage],
                functions: [Components.Schemas.ChatCompletionFunctions]? = nil,
                function_call: Components.Schemas.CreateChatCompletionRequest
                    .function_callPayload? = nil,
                temperature: Swift.Double? = nil,
                top_p: Swift.Double? = nil,
                n: Swift.Int? = nil,
                stream: Swift.Bool? = nil,
                stop: Components.Schemas.CreateChatCompletionRequest.stopPayload? = nil,
                max_tokens: Swift.Int? = nil,
                presence_penalty: Swift.Double? = nil,
                frequency_penalty: Swift.Double? = nil,
                logit_bias: Components.Schemas.CreateChatCompletionRequest.logit_biasPayload? = nil,
                user: Swift.String? = nil
            ) {
                self.messages = messages
                self.functions = functions
                self.function_call = function_call
                self.temperature = temperature
                self.top_p = top_p
                self.n = n
                self.stream = stream
                self.stop = stop
                self.max_tokens = max_tokens
                self.presence_penalty = presence_penalty
                self.frequency_penalty = frequency_penalty
                self.logit_bias = logit_bias
                self.user = user
            }
            public enum CodingKeys: String, CodingKey {
                case messages
                case functions
                case function_call
                case temperature
                case top_p
                case n
                case stream
                case stop
                case max_tokens
                case presence_penalty
                case frequency_penalty
                case logit_bias
                case user
            }
        }
```

It seems that parsing the `anyOf` field is skipped. If I replaced the `anyOf` with `type: string`, `model` key is appeared in the generated code. 

Is the `anyOf` supported right now? Or are there any workarounds?

**Comment by czechboy0** [comment]
Hi @SokichiFujita – if it's missing, it might have been skipped. The generated should have emitted a warning diagnostic about it - did you see one?

Let me see if I can reproduce the issue and get back to you.

**Comment by czechboy0** [comment]
So, I tried to reproduce it, and it works correctly on the latest version 0.1.9: https://github.com/apple/swift-openapi-generator/releases/tag/0.1.9

@SokichiFujita - please try again with the latest version of the generator and close this issue if that resolves the issue, thanks 🙏

**Comment by SokichiFujita** [comment]
@czechboy0 

Oh, you just released the 0.1.9 yesterday! 

I used the 0.1.8 but after updating the 0.1.9, `anyOf` skip issue has solved completely! Thank you.

Following is the new `Types.swift`

```
        /// - Remark: Generated from `#/components/schemas/CreateChatCompletionRequest`.
        public struct CreateChatCompletionRequest: Codable, Equatable, Hashable, Sendable {
            /// ID of the model to use. See the [model endpoint compatibility](/docs/models/model-endpoint-compatibility) table for details on which models work with the Chat API.
            ///
            /// - Remark: Generated from `#/components/schemas/CreateChatCompletionRequest/model`.
            public struct modelPayload: Codable, Equatable, Hashable, Sendable {
                /// - Remark: Generated from `#/components/schemas/CreateChatCompletionRequest/model/value1`.
                public var value1: Swift.String?
                /// - Remark: Generated from `#/components/schemas/CreateChatCompletionRequest/model/value2`.
                @frozen
                public enum Value2Payload: RawRepresentable, Codable, Equatable, Hashable, Sendable,
                    _AutoLosslessStringConvertible, CaseIterable
                {
                    case gpt_4
                    case gpt_4_0314
                    case gpt_4_0613
                    case gpt_4_32k
                    case gpt_4_32k_0314
                    case gpt_4_32k_0613
                    case gpt_3_5_turbo
                    case gpt_3_5_turbo_16k
                    case gpt_3_5_turbo_0301
                    case gpt_3_5_turbo_0613
                    case gpt_3_5_turbo_16k_0613
                    /// Parsed a raw value that was not defined in the OpenAPI document.
                    case undocumented(String)
                    public init?(rawValue: String) {
                        switch rawValue {
                        case "gpt-4": self = .gpt_4
                        case "gpt-4-0314": self = .gpt_4_0314
                        case "gpt-4-0613": self = .gpt_4_0613
                        case "gpt-4-32k": self = .gpt_4_32k
                        case "gpt-4-32k-0314": self = .gpt_4_32k_0314
                        case "gpt-4-32k-0613": self = .gpt_4_32k_0613
                        case "gpt-3.5-turbo": self = .gpt_3_5_turbo
                        case "gpt-3.5-turbo-16k": self = .gpt_3_5_turbo_16k
                        case "gpt-3.5-turbo-0301": self = .gpt_3_5_turbo_0301
                        case "gpt-3.5-turbo-0613": self = .gpt_3_5_turbo_0613
                        case "gpt-3.5-turbo-16k-0613": self = .gpt_3_5_turbo_16k_0613
                        default: self = .undocumented(rawValue)
                        }
                    }
                    public var rawValue: String {
                        switch self {
                        case let .undocumented(string): return string
                        case .gpt_4: return "gpt-4"
                        case .gpt_4_0314: return "gpt-4-0314"
                        case .gpt_4_0613: return "gpt-4-0613"
                        case .gpt_4_32k: return "gpt-4-32k"
                        case .gpt_4_32k_0314: return "gpt-4-32k-0314"
                        case .gpt_4_32k_0613: return "gpt-4-32k-0613"
                        case .gpt_3_5_turbo: return "gpt-3.5-turbo"
                        case .gpt_3_5_turbo_16k: return "gpt-3.5-turbo-16k"
                        case .gpt_3_5_turbo_0301: return "gpt-3.5-turbo-0301"
                        case .gpt_3_5_turbo_0613: return "gpt-3.5-turbo-0613"
                        case .gpt_3_5_turbo_16k_0613: return "gpt-3.5-turbo-16k-0613"
                        }
                    }
                    public static var allCases: [Value2Payload] {
                        [
                            .gpt_4, .gpt_4_0314, .gpt_4_0613, .gpt_4_32k, .gpt_4_32k_0314,
                            .gpt_4_32k_0613, .gpt_3_5_turbo, .gpt_3_5_turbo_16k,
                            .gpt_3_5_turbo_0301, .gpt_3_5_turbo_0613, .gpt_3_5_turbo_16k_0613,
                        ]
                    }
                }
                /// - Remark: Generated from `#/components/schemas/CreateChatCompletionRequest/model/value2`.
                public var value2:
                    Components.Schemas.CreateChatCompletionRequest.modelPayload.Value2Payload?
                /// Creates a new `modelPayload`.
                ///
                /// - Parameters:
                ///   - value1:
                ///   - value2:
                public init(
                    value1: Swift.String? = nil,
                    value2: Components.Schemas.CreateChatCompletionRequest.modelPayload
                        .Value2Payload? = nil
                ) {
                    self.value1 = value1
                    self.value2 = value2
                }
                public init(from decoder: any Decoder) throws {
                    value1 = try? .init(from: decoder)
                    value2 = try? .init(from: decoder)
                    try DecodingError.verifyAtLeastOneSchemaIsNotNil(
                        [value1, value2],
                        type: Self.self,
                        codingPath: decoder.codingPath
                    )
                }
                public func encode(to encoder: any Encoder) throws {
                    try value1?.encode(to: encoder)
                    try value2?.encode(to: encoder)
                }
            }

```

**Comment by SokichiFujita** [comment]
I confirmed that the issue is solved with latest 0.1.9.
